modules/core/keeper/msg_server.go

```diff
@@ -503,6 +503,7 @@ func (k Keeper) RecvPacket(goCtx context.Context, msg *channeltypes.MsgRecvPacke
 	if ack == nil || ack.Success() {
 		// write application state changes for asynchronous and successful acknowledgements
 		writeFn()
+		ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
```

Comment:
could you add a comment that indicates that `ctx.CacheContext()` sets a new event manager and thus won't cause duplicate events to be returned by `cacheCtx.EventManager().Events()`

At first I thought this line might cause events emitted by `k.ChannelKeeper.RecvPacket` to be represented twice but I realized the cache context reset the events by setting a new event manager

It'd be great to have this written in the code